codewind-che-sidecar/tests/cronjob-sidecar.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v20.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+
+CODEWIND_CHE_PLUGIN_DIR=~/codewind-che-plugin
+CODEWIND_CHE_PLUGIN_TEST_DIR=$CODEWIND_CHE_PLUGIN_DIR/codewind-che-sidecar/tests
+CODEWIND_CHE_PLUGIN_REPO=git@github.com:eclipse/codewind-che-plugin.git
+TEST_BRANCH="master"
+
+DATE_NOW=$(date +"%d-%m-%Y")
+TIME_NOW=$(date +"%H.%M.%S")
+BUCKET_NAME=codewind-che-plugin-sidecar
+TEST_OUTPUT_DIR=~/test_results/sidecar/$DATE_NOW/$TIME_NOW
+TEST_OUTPUT_TAP=$TEST_OUTPUT_DIR/test_output.tap
+TEST_OUTPUT_XML=$TEST_OUTPUT_DIR/test_output.xml
+
+if [[ (-z $NAMESPACE) ]]; then
+  echo -e "${RED}Mandatory argument NAMESPACE is not set up. ${RESET}\n"
+  echo -e "${RED}Please export variable NAMESPACE to run the Codewind sidecar tests. ${RESET}\n"
+  exit 1
+fi
+if [[ (-z $CLUSTER_IP) ]]; then
+  echo -e "${RED}Mandatory argument CLUSTER_IP is not set up. ${RESET}\n"
+  echo -e "${RED}Please export variable CLUSTER_IP to run the Codewind sidecar tests. ${RESET}\n"
+  exit 1
+fi
+
+if [[ (-z $CLUSTER_USER) ]]; then
+  echo -e "${RED}Mandatory argument CLUSTER_USER is not set up. ${RESET}\n"
+  echo -e "${RED}Please export variable CLUSTER_USER to run the Codewind sidecar tests. ${RESET}\n"
+  exit 1
+fi
+
+if [[ (-z $CLUSTER_PASSWORD) ]]; then
+  echo -e "${RED}Mandatory argument CLUSTER_PASSWORD is not set up. ${RESET}\n"
+  echo -e "${RED}Please export variable CLUSTER_PASSWORD to run the Codewind sidecar tests. ${RESET}\n"
+  exit 1
+fi
+if [[ (-z $DASHBOARD_IP) ]]; then
+  echo -e "${RED}Dashboard IP is required to upload test results. ${RESET}\n"
+  exit 1
+fi
+
+oc login $CLUSTER_IP:8443 -u $CLUSTER_USER -p $CLUSTER_PASSWORD
+if [[ $? -eq 0 ]]; then
+  echo -e "${GREEN}Successfully logged into the OKD cluster ${RESET}\n"
+else
+  echo -e "${RED}Failed to log into the OKD cluster ${RESET}\n"
+  exit 1
+fi
+
+export CHE_INGRESS_DOMAIN=che-$NAMESPACE.$CLUSTER_IP.nip.io
```

Comment:
Nit: We should probably do as we did for the other test scripts and just pass in the entire ingress domain as an arg, rather than the namespace and cluster IP separately. 

This'll help with eventual testing on OS4, which doesn't use .nip.io, or in test clusters where the cluster IP doesn't match the ingress domain.

Reply:
@johnmcollier agreed.. so that's why the $CHE_INGRESS_DOMAIN is what is used by the side car tests for external users. For our currently nightly pfe/turbine/etc runs, the way our scripts are setup the CLUSTER_IP is set as a separate global env var. So given that this cronjob-sidecar.sh is mostly for our own nightly-run use anyways, maybe not a big issue for now, but we should change for the future.